From 34066eb245dc61e03716282e1141218833368ded Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:34 -0400
Subject: [usb] uas: Move uas_find_endpoints to uas-detect.h

Message-id: <1410887463-20674-323-git-send-email-dzickus@redhat.com>
Patchwork-id: 95608
O-Subject: [RHEL7 PATCH 322/591] uas: Move uas_find_endpoints to uas-detect.h
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d77adc0284beea5783c52b2af49217a37c2114cd
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Tue Oct 29 10:03:34 2013 +0100

    uas: Move uas_find_endpoints to uas-detect.h

    No changes, just the move.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/storage/uas-detect.h b/drivers/usb/storage/uas-detect.h
index 082bde1..8de030a 100644
--- a/drivers/usb/storage/uas-detect.h
+++ b/drivers/usb/storage/uas-detect.h
@@ -40,6 +40,33 @@ static int uas_find_uas_alt_setting(struct usb_interface *intf)
  return -ENODEV;
 }
 
+static int uas_find_endpoints(struct usb_host_interface *alt,
+         struct usb_host_endpoint *eps[])
+{
+ struct usb_host_endpoint *endpoint = alt->endpoint;
+ unsigned i, n_endpoints = alt->desc.bNumEndpoints;
+
+ for (i = 0; i < n_endpoints; i++) {
+  unsigned char *extra = endpoint[i].extra;
+  int len = endpoint[i].extralen;
+  while (len >= 3) {
+   if (extra[1] == USB_DT_PIPE_USAGE) {
+    unsigned pipe_id = extra[2];
+    if (pipe_id > 0 && pipe_id < 5)
+     eps[pipe_id - 1] = &endpoint[i];
+    break;
+   }
+   len -= extra[0];
+   extra += extra[0];
+  }
+ }
+
+ if (!eps[0] || !eps[1] || !eps[2] || !eps[3])
+  return -ENODEV;
+
+ return 0;
+}
+
 static int uas_use_uas_driver(struct usb_interface *intf,
          const struct usb_device_id *id)
 {
diff --git a/drivers/usb/storage/uas.c b/drivers/usb/storage/uas.c
index 7662b3e..5cedc7f 100644
--- a/drivers/usb/storage/uas.c
+++ b/drivers/usb/storage/uas.c
@@ -898,33 +898,6 @@ static int uas_switch_interface(struct usb_device *udev,
    intf->altsetting[0].desc.bInterfaceNumber, alt);
 }
 
-static int uas_find_endpoints(struct usb_host_interface *alt,
-         struct usb_host_endpoint *eps[])
-{
- struct usb_host_endpoint *endpoint = alt->endpoint;
- unsigned i, n_endpoints = alt->desc.bNumEndpoints;
-
- for (i = 0; i < n_endpoints; i++) {
-  unsigned char *extra = endpoint[i].extra;
-  int len = endpoint[i].extralen;
-  while (len >= 3) {
-   if (extra[1] == USB_DT_PIPE_USAGE) {
-    unsigned pipe_id = extra[2];
-    if (pipe_id > 0 && pipe_id < 5)
-     eps[pipe_id - 1] = &endpoint[i];
-    break;
-   }
-   len -= extra[0];
-   extra += extra[0];
-  }
- }
-
- if (!eps[0] || !eps[1] || !eps[2] || !eps[3])
-  return -ENODEV;
-
- return 0;
-}
-
 static void uas_configure_endpoints(struct uas_dev_info *devinfo)
 {
  struct usb_host_endpoint *eps[4] = { };
-- 
1.7.1